test/extensions/filters/common/rbac/matchers_test.cc

```diff
@@ -290,6 +290,23 @@ TEST(PolicyMatcher, PolicyMatcher) {
   checkMatcher(matcher, false, conn);
 }
 
+TEST(RequestedServerNameMatcher, ValidRequestedServerName) {
+  Envoy::Network::MockConnection conn;
+  EXPECT_CALL(conn, requestedServerName())
+      .Times(2)
+      .WillRepeatedly(Return(absl::string_view("www.cncf.io")));
+
+  checkMatcher(RequestedServerNameMatcher("www.cncf.io"), true, conn);
+  checkMatcher(RequestedServerNameMatcher("example.com"), false, conn);
```

Comment:
and based on the feedback above, would be good to have prefix/exact/regex match tests

Reply:
Addressed in 527170f.